pkg/services/user/userimpl/userimpl.go

```diff
@@ -0,0 +1,91 @@
+package userimpl
+
+import (
+	"context"
+	"time"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/org"
+	"github.com/grafana/grafana/pkg/services/sqlstore/db"
+	"github.com/grafana/grafana/pkg/services/user"
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/grafana/grafana/pkg/util"
+)
+
+type Service struct {
+	store      store
+	orgService org.Service
+}
+
+func ProvideService(db db.DB) user.Service {
+	return &Service{
+		store: &sqlStore{
+			db: db,
+		},
+	}
+}
+
+func (s *Service) Create(ctx context.Context, cmd *user.CreateUserCommand) (*user.User, error) {
+	orgID, err := s.orgService.GetIDForNewUser(*cmd)
+	if err != nil {
+		return nil, err
+	}
+
+	if cmd.Email == "" {
+		cmd.Email = cmd.Login
+	}
+	cmd.OrgID = orgID
+	usr, err := s.store.Get(ctx, cmd)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I think that if the user exists then this it should fail: that's what happens in the existing implementation:
https://github.com/grafana/grafana/blob/32d4f6ac60f3169ddb3128d55145cad644fa7a3e/pkg/services/sqlstore/user.go#L53-L55